src/services/genericSettingsApi.ts

```diff
@@ -21,7 +21,7 @@ export const getAnnouncementBanner = (requestConfig: AxiosRequestConfig = {}) =>
  * @param requestConfig - Axios based request config
  */
 export const setAnnouncementBanner = ({ message }: { message: string }, requestConfig: AxiosRequestConfig = {}) => {
-  return pncClient.getHttpClient().post<void>('/generic-setting/announcement-banner', message, requestConfig);
+  return pncClient.getHttpClient().post<undefined>('/generic-setting/announcement-banner', message, requestConfig);
```

Comment:
This absolutely IS valid.

Reply:
@patrikk0123 please provide additional sources

Reply:
Firstly, even if it was not quite valid, it's needed, since data structure (such as data of service container) cannot be of void type. Secondly, the semantics of returning void and undefined are pretty much the samee (it returns nothing). There is no JSON format consisting of just `undefined` JS value.

Here is some distinction provided by ChatGPT:

```
void: the return type is explicitly marked as void, indicating that you are not interested in the response data.
```

```
undefined: it's a matter of personal preference and may be considered a bit more explicit, although it essentially conveys the same idea.
```

Reply:
done